search-commons/src/main/java/no/unit/nva/search/ticket/TicketAccessFilter.java

```diff
@@ -0,0 +1,276 @@
+                .add(new TermQueryBuilder(OWNER_USERNAME, currentUser))
+                .add(new TermsQueryBuilder(TYPE_KEYWORD, allowedTicketTypes))
+                .queryName(ANY_OF_TICKET_TYPE_OR_USER_NAME);
+    }
+
+    protected QueryBuilder getOrgFilter() {
+        return new AcrossFieldsQuery<TicketParameter>()
+                .buildQuery(ORGANIZATION_ID, organizationId.toString())
+                .findFirst()
+                .orElseThrow()
+                .getValue()
+                .queryName(ORGANIZATION_ID.asCamelCase());
+    }
+
+    protected BoolQueryBuilder getExcludedFilter() {
+        var excludedSet =
+                nonNull(excludeTicketTypes)
+                        ? excludeTicketTypes
+                        : Arrays.stream(TicketType.values())
+                                .filter(f -> !allowedTicketTypes.contains(f))
+                                .collect(
+                                        Collectors.toCollection(
+                                                () -> EnumSet.noneOf(TicketType.class)));
+        return boolQuery()
+                .mustNot(termsQuery(TYPE_KEYWORD, excludedSet))
+                .queryName(NOT_ANY_OF_TICKET_TYPE);
+    }
+
+    private TicketAccessFilter includedTypes(Set<TicketType> ticketTypes) {
+        this.allowedTicketTypes = ticketTypes;
+        return this;
+    }
+
+    private TicketAccessFilter excludeTypes(Set<TicketType> ticketTypes) {
+        this.excludeTicketTypes = ticketTypes;
+        return this;
+    }
+
+    /**
+     * Apply business rules to determine which ticket types are allowed.
+     *
+     * <ul>
+     *   <lh>Business Rules.</lh>
+     *   <li>manage_doi -> DOI_REQUEST
+     *   <li>support -> GENERAL_SUPPORT_CASE
+     *   <li>manage_publishing_requests -> PUBLISHING_REQUEST
+     *   <li>manage_customers -> DOI_REQUEST, GENERAL_SUPPORT_CASE, PUBLISHING_REQUEST
+     *   <li>is_owner -> DOI_REQUEST, GENERAL_SUPPORT_CASE, PUBLISHING_REQUEST
+     * </ul>
+     */
+    private Set<TicketType> applyBusinessRules(List<AccessRight> accessRights)
+            throws UnauthorizedException {
+
+        var isOwner = searchAsTicketOwner() && validateOwner(currentUser);
+        var isCustomManager = searchAsCustomManager() && validateCustomManager(accessRights);
+
+        var allowed = EnumSet.noneOf(TicketType.class);
+        if (accessRights.contains(MANAGE_DOI) || isOwner || isCustomManager) {
+            allowed.add(TicketType.DOI_REQUEST);
+        }
+        if (accessRights.contains(AccessRight.SUPPORT) || isOwner || isCustomManager) {
+            allowed.add(TicketType.GENERAL_SUPPORT_CASE);
+        }
+        if (accessRights.contains(MANAGE_PUBLISHING_REQUESTS) || isOwner || isCustomManager) {
+            allowed.add(TicketType.PUBLISHING_REQUEST);
+        }
+        if (allowed.isEmpty()) {
+            throw new UnauthorizedException(USER_HAS_NO_ACCESS_TO_SEARCH_FOR_ANY_TICKET_TYPES);
+        }
```

Comment:
Maybe common check can be extracted in each if statement -> `doSomething(ACCESS_RIGHT)`

Reply:
This procedure is the core of the accesslogic to Tickets.
It needs to be all there, no obscuring function that hides conditions and values.
It took me a very long time to get an overview of what rules were in play.

That is the reason for doing it this way.